bigtable-client-core-parent/bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/async/TestBulkMutationGCJClient.java

```diff
@@ -0,0 +1,184 @@
+public class TestBulkMutationGCJClient {
+
+  @Rule
+  public ExpectedException expect = ExpectedException.none();
+
+  @Mock private UnaryCallable<RowMutation, Void> callable;
+  private BulkMutationGCJClient bulkMutationClient;
+  private SettableApiFuture<Void> future;
+  private RowMutation rowMutation = RowMutation.create("fake-table", "fake-key");
+  private final BulkMutationBatcher batcher = new BulkMutationBatcher(new BatcherUnaryCallable());
+
+  @Before
+  public void setUp() {
+    BulkMutationBatcher bulkMutationBatcher = new BulkMutationBatcher(callable);
+    bulkMutationClient = new BulkMutationGCJClient(bulkMutationBatcher);
+    future = SettableApiFuture.create();
+  }
+
+  @Test
+  public void testAdd() {
+    when(callable.futureCall(rowMutation)).thenReturn(future);
+    ApiFuture<Void> result = bulkMutationClient.add(rowMutation);
+    Assert.assertFalse(result.isDone());
+    future.set(null);
+    Assert.assertTrue(result.isDone());
+    verify(callable).futureCall(rowMutation);
+  }
+
+  @Test(expected = ExecutionException.class)
+  public void testAddFailure() throws Exception{
+    String message = "can not perform mutation";
+    when(callable.futureCall(rowMutation)).thenReturn(future);
+    future.setException(new RuntimeException(message));
+    ApiFuture<Void> result = bulkMutationClient.add(rowMutation);
+    Assert.assertTrue(result.isDone());
+    result.get();
+    expect.expectMessage(message);
+    verify(callable).futureCall(rowMutation);
+  }
+
+  @Test
+  public void testCallableTooFewStatuses() {
+    RowMutation rowMutationOther = RowMutation.create("table", "key");
+    SettableApiFuture<Void> futureOther = SettableApiFuture.create();
+    when(callable.futureCall(rowMutation)).thenReturn(future);
+    when(callable.futureCall(rowMutationOther)).thenReturn(futureOther);
+
+    ApiFuture<Void> result = bulkMutationClient.add(rowMutation);
+    ApiFuture<Void> resultOther = bulkMutationClient.add(rowMutationOther);
+    // Only resolving one response
+    future.set(null);
+
+    Assert.assertTrue(result.isDone());
+    Assert.assertFalse(resultOther.isDone());
+
+    Assert.assertFalse(bulkMutationClient.isFlushed());
+    futureOther.set(null);
+    Assert.assertTrue(bulkMutationClient.isFlushed());
+
+    verify(callable).futureCall(rowMutation);
+    verify(callable).futureCall(rowMutationOther);
+  }
+
+  @Test
+  public void testConcurrentBatches() throws Exception {
+    final List<ApiFuture<Void>> futures =
+        Collections.synchronizedList(new ArrayList<ApiFuture<Void>>());
+    final int batchCount = 10;
+    final int concurrentBulkMutationCount = 50;
+
+    Runnable r = new Runnable() {
+      @Override
+      public void run() {
+        IBulkMutation bulkMutation = new BulkMutationGCJClient(batcher);
+        for (int i = 0; i < batchCount * 10; i++) {
+          futures.add(bulkMutation.add(RowMutation.create("fake-table", "fake-key")));
+        }
+        bulkMutation.sendUnsent();
```

Comment:
`sendUnsent()` doesn't actually do anything.  Why use it here.

Reply:
Thanks for the comment. Again I used it for initial implementation now I have removed it.